client/src/pages/Editor.tsx

```diff
@@ -0,0 +1,236 @@
+                    xml: diagramData.current, 
+                    title: diagramNameRef.current, 
+                    challengeId: query.get("challengeId") //NOTE: This only works for challenges in progress
+                }),
+                keepalive: true // To allow saves when the tab is closed
+            })
+            .then(response => {
+                if (!response.ok) {
+                    throw new Error(response.statusText);
+                }
+                return response.json() as Promise<{id: string}>
+            })
+            .then(data => {
+                // console.log('Success:', data);
+            })
+            .catch((error) => {
+                console.error('Error:', error);
+            });
+        }
+    },[query, diagramName]);
+
+
+    return (
+        <div>
+            <h1>{diagramName}</h1>
+            <div style={{height: "100vh"}}>
+                < DrawIoEmbed 
+                    ref={drawioRef}
+                    configuration={CONFIG}
+                    urlParameters={
+                        {
+                            // ui: "atlas",
+                            spin: true,
+                            modified: true,
+                            keepmodified: true,
+                            libraries: true,
+                            noSaveBtn: true,
+                            saveAndExit: false,
+                            noExitBtn: true
+                        }
+                    }
+                    onExport={(data) => {handleSave(data)}} //Note: it says the type is wrong, but it is not. 
+                    onLoad={(data) => {
+                        // If the xml is null, then we need to wait and reload the diagram from diagramData
+                        if (data.xml === null) {
+                            console.log("Waiting for diagramData to be initialized");
+                            setTimeout(() => {
+                                if (drawioRef.current !== null && diagramData.current !== undefined){
+                                    drawioRef.current?.load({
+                                        xmlpng: diagramData.current
+                                    });
+                                    console.log("Loaded diagram from stash");
+                                }
+                                else {
+                                    console.log("No diagram data found");
+                                }
+                            }, 100);
+                        }  
+                    }}
+                    onClose={(data) => {
+                        // Save and exit. Timeout to make sure the save request is sent before the tab is closed.
+                        doExport();
+                        setTimeout(() => {
+                            window.close();
+                        }, 1000);
+                    }}
+                    />
+            </div>
+            {/* TESTING ONLY
```

Comment:
Is it intentional to keep this commented block?

Reply:
Yes